### PR TITLE
CI: add python 3.10, 3.11, 3.12 build

### PR DESCRIPTION
**Pull request by jrfk**
> [!NOTE]
> This PR is a redo of https://github.com/groove-x/trio-util/pull/25.

## What
This PR adds CI build configurations for Python versions 3.10, 3.11, and 3.12.

## Why
To ensure our project remains compatible with newer Python versions, improving user experience and future-proofing our codebase.

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,33 +1,34 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.8
+# by the following command:
```
Comment (belm0) [inline test-requirements.txt:3]: What version of pip-tools did you use to generate this file?

This requirements file installs `pip-tools==6.2.0`.  If you used a different version, it might cause formatting changes like this.

File: test-requirements.txt [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
 #
-#    pip-compile --output-file=test-requirements.txt setup.py test-requirements.in
+#    pip-compile -q -P pytest --output-file=test-requirements.txt setup.py test-requirements.in
```
Comment (belm0) [inline test-requirements.txt:5]: (revert this change)

File: test-requirements.txt (hunk 1 above)
Comment (jrfk) [reply test-requirements.txt:3]: I was using `pip-tools==7.4.1`. I've checked the related packages and updated the test-requirements.txt. [62dca6b](https://github.com/groove-x/trio-util/pull/26/commits/62dca6bb265f3c21b9296404d0d19e5bfe587b00)

File: setup.py [hunk 3]
```diff
@@ -47,6 +47,9 @@
         'Programming Language :: Python :: 3.7',
```
Comment (belm0) [inline setup.py:47]: actually, someone reminded me that python 3.7 is past end of life

**Comment by jrfk** [comment]
Thank you!
